src/software/sensor_fusion/filter/possession_filter.cpp

```diff
@@ -0,0 +1,49 @@
+#include "software/sensor_fusion/filter/possession_filter.h"
+
+std::vector<RobotIdWithTeamSide> getRobotsWithPossession(
+    std::vector<RobotId> friendly_robots_with_breakbeam_triggered, Team friendly_team,
+    Team enemy_team, Ball ball)
+{
+    std::vector<RobotIdWithTeamSide> possessions;
+    for (const auto &robot_id : friendly_robots_with_breakbeam_triggered)
+    {
+        possessions.push_back(
+            RobotIdWithTeamSide{.id = robot_id, .team_side = TeamSide::FRIENDLY});
+    }
+
+    for (const auto &robot : friendly_team.getAllRobots())
+    {
+        if (ballNearDribbler(ball.position(), robot.position(), robot.orientation()))
+        {
+            possessions.push_back(
+                RobotIdWithTeamSide{.id = robot.id(), .team_side = TeamSide::FRIENDLY});
+        }
+    }
+
+    for (const auto &robot : enemy_team.getAllRobots())
+    {
+        if (ballNearDribbler(ball.position(), robot.position(), robot.orientation()))
+        {
+            possessions.push_back(
+                RobotIdWithTeamSide{.id = robot.id(), .team_side = TeamSide::ENEMY});
+        }
+    }
+
+    return possessions;
+}
+
+bool ballNearDribbler(Point ball_position, Point robot_position, Angle robot_orientation)
+{
+    static const double POSSESSION_THRESHOLD_METERS = ROBOT_MAX_RADIUS_METERS + 0.2;
```

Comment:
`0.2` seems arbitrary. Any reason it was chosen? Also should this account for ball radius?

Reply:
This was copied over from the equivalent evaluation function

Reply:
That's probably best for now

Reply:
I think this will be dealt with in https://github.com/UBC-Thunderbots/Software/pull/1619/